react/features/notifications/components/AbstractNotificationsContainer.js

```diff
@@ -51,18 +50,40 @@ export default class AbstractNotificationsContainer<P: Props>
         this._onDismissed = this._onDismissed.bind(this);
     }
 
+    /**
+     * Sets a timeout for the first notification (if applicable).
+     *
+     * @inheritdoc
+     */
+    componentDidMount() {
+        // Set the initial dismiss timeout (if any)
+        this._manageDismissTimeout();
+    }
+
     /**
      * Sets a timeout if the currently displayed notification has changed.
      *
      * @inheritdoc
      */
     componentDidUpdate(prevProps: P) {
+        this._manageDismissTimeout(prevProps);
+    }
+
+    /**
+     * Sets/clears the dismiss timeout for the top notification.
+     *
+     * @param {P} [prevProps] - The previous properties (if called from
```

Comment:
Why the bracket? Is it needed for something?

Reply:
It's the optional parameter's syntax:
http://usejsdoc.org/tags-param.html#optional-parameters-and-default-values

Reply:
But is it really an optional parameter, or just a nullable one? Because if it's a nullable one, then it's not the correct annotation. Then it should be ```@param {?P} prevProps``` or something

Reply:
ok

Reply:
isn't the `?` sign specific to flow ?

Reply:
Yeah I think that the `?` applies to flow types.
In JSDoc a nullable can be either `[something]` or `{number|null}`. So because I don't care whether it's `null` or `undefined` I'm thinking that `[something]` is easier than '{something|null|undefined}`